scripts/renumber_ants.py

```diff
@@ -0,0 +1,330 @@
+#! /usr/bin/env python
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2016 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""
+Script to handle the ONE TIME renumbering of HERA dishes.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+from hera_mc import mc, cm_utils, part_connect, cm_handling, geo_handling, cm_hookup, cm_part_revisions
+import sys
+import copy
+
+
+def query_renumber(args):
+    """
+    Gets renumber information from user
+    """
+    if args.station_name == None:
+        args.station_name = raw_input('Station name for which to renumber:  ')
+    args.date = cm_utils._query_default('date', args)
+    return args
+
+
+def stop_previous_parts(args,antrev,feedrev):
+    """This adds stop times to the previous parts"""
+    current = int(args.date.gps)
+    args.add_new_part = False
+
+    print("Stopping part %s %s at %s" % (antrev[0], antrev[1], str(args.date)))
+    data = [[antrev[0], antrev[1], 'stop_gpstime', current]]
+    
+    if feedrev[1] == 'A':
+        print("Stopping part %s %s at %s" % (feedrev[0], feedrev[1], str(args.date)))
+        data.append([feedrev[0], feedrev[1], 'stop_gpstime', current])
+
+    if args.actually_do_it:
+        part_connect.update_part(args, data)
+    else:
+        print(data)
+
+
+def add_new_parts(args,antrev,mfgna,feedrev):
+    """This adds the new rev B antenna and FDA rev B"""
+    current = int(args.date.gps)
+    args.add_new_part = True
+
+    print("Adding part %s %s at %s" % (antrev[0], antrev[1], str(args.date)))
+    data =     [[antrev[0], antrev[1], 'hpn', antrev[0]]]
+    data.append([antrev[0], antrev[1], 'hpn_rev', antrev[1]])
+    data.append([antrev[0], antrev[1], 'hptype', 'antenna'])
+    data.append([antrev[0], antrev[1], 'manufacturer_number', mfgna])
+    data.append([antrev[0], antrev[1], 'start_gpstime', current])
+
+    if feedrev[1] == 'A':
+        mfgnf = get_mfg_number(feedrev)
+        print("Adding part %s %s at %s" % (feedrev[0], 'B', str(args.date)))
+        data.append([feedrev[0], 'B', 'hpn', feedrev[0]])
+        data.append([feedrev[0], 'B', 'hpn_rev', 'B'])
+        data.append([feedrev[0], 'B', 'hptype', 'feed'])
+        data.append([feedrev[0], 'B', 'manufacturer_number', mfgnf])
+        data.append([feedrev[0], 'B', 'start_gpstime', current])
+
+    if args.actually_do_it:
+        part_connect.update_part(args, data)
+    else:
+        print(data)
+
+def get_connection_key(c,p):
+    for ck in c.keys():
+        if p[0] in ck:
+            break
```

Comment:
Not important since this is a one-time script that has presumably already served its purpose, but you could just to `return ck` here, and `return None` if the `for` loop exits.